### DISCUSSION TITLE
Advice Needed: Transitioning from table data regression models to GNN

### DISCUSSION THREAD

**Discussion by heaynking** [pyg-team/pytorch_geometric#9406] [Q&A]
Hi,

I'm currently working on transitioning from using regression models with table data to exploring Graph Neural Networks (GNNs).
This is my first time working with graph networks, and I would greatly appreciate any advice or insights you can provide.

For example, suppose we have the following four tables.

ID | Name | Location | Phone | Target
-- | -- | -- | -- | --
0 | L.H | A3 | B2 | 0.8
1 | H.R | A2 | B1 | 0.7
… | … | … | … | …
N | T.S | A4 | B6 | 1.0

| ID  | Feature1 | Feature2 |
| --- | -------- | -------- |
| L.H | 54       | 43       |
| H.R | 156      | 36       |
| …   | …        | …        |
| T.S | 14       | 26       |

| ID | A-Feature1 | A-Feature2 | A-Feature3 |
| -- | ---------- | ---------- | ---------- |
| A1 | 3          | 1          | 177        |
| A2 | 6          | 5          | 244        |
| A3 | 4          | 5          | 453        |
| A4 | 19         | 8          | 132        |

| ID | B-Feature1 | B-Feature2 |
| -- | ---------- | ---------- |
| B1 | 400        | 36         |
| B2 | 300        | 46         |
| …  | …          | …          |
| B3 | 145        | 51         |

There are more columns and rows in the data, but I simplified it to illustrate.
I want to use the relationships between these to create graph data and perform a task to predict the target for new data.
The new data will be a combination of the items listed in the above tables.

---

First, I would like advice on what kind of graph data and tasks should be set.
Currently, I'm considering the following two:

1. It is a task of making predictions at the **graph level**.

![image](https://github.com/pyg-team/pytorch_geometric/assets/25063447/280ac756-0fe3-4fe0-8114-01be276dc158)

2. The other is the task of making predictions at the **node level**.

![image](https://github.com/pyg-team/pytorch_geometric/assets/25063447/172ed109-0c1b-44c3-a48a-3d4d460d4f6e)

 As shown in the table, the number of features is different, so that's a point of concern that should be noted.

I know the model's performance depends on the data's quality or the architecture itself, but I would appreciate it if we could discuss which approach is more effective in such cases.

**Comment by rusty1s** [comment]
What do you mean by the number of features is different? As long as they are consistent on the table-level, you should be absolutely fine. This is a perfect use-case for PyG, and we have tried to write many resources on it:
* https://pytorch-geometric.readthedocs.io/en/latest/tutorial/load_csv.html
* https://arxiv.org/abs/2312.04615

**Reply by heaynking** [reply]
Thank you very much for the answer!!
I mean that the number of node features is different. So, It should be heterogeneous data.

What do you think about the regression task? 
Is it ok to use a graph-level prediction task? Or should I use a node-level prediction task instead? 
Are there any guidelines or good ways to determine this?

**Reply by rusty1s** [reply]
From the example, it looks like this is a node-level prediction task since a target is attached to a specific node. Let me know if I am misunderstanding.
